cli/templates/CodeGen.Generate.elm.ts

```diff
@@ -1,91 +1,169 @@
 export default () =>
   `
-port module Gen.CodeGen.Generate exposing (run, File, error, files, info)
+port module Gen.CodeGen.Generate exposing
+    ( noInput, simple, simpleFrom
+    , run
+    , File
+    , error, files, info
+    )
 
-import Json.Encode as Json
+{-|
 
 
-run : (flags -> List File) -> Program flags () ()
-run f =
-     Platform.worker
-          { init =
-               \\flags ->
-                    ( ()
-                    , files (f flags)
-                    )
-          , update =
-               \\_ model ->
-                    ( model, Cmd.none )
-          , subscriptions = \\_ -> Sub.none
-          }
+# Simple API
 
+@docs noInput, simple, simpleFrom
 
-type alias File =
-     { path : String
-     , contents : String
-     }
 
+# With errors
 
-encodeFile : File -> Json.Value
-encodeFile file =
-     Json.object
-          [ ( "path", Json.string file.path )
-          , ( "contents", Json.string file.contents )
-          ]
+@docs run, Error
 
 
-{-|
+# Types
+
+@docs File
+
+
+# Flexible API
 
-     Provide the list of files to be generated.
-     These files will be generated and the script will end.
+This is the low level API if you need more control over the generation process proceeds.
+
+@docs error, files, info
 
 -}
-files : List File -> Cmd msg
-files list =
-     onSuccessSend (List.map encodeFile list)
 
+import Json.Decode exposing (Decoder)
+import Json.Encode as Json
+
+
+{-| Produce a list of files without taking input.
+-}
+noInput : List File -> Program flags () ()
```

Comment:
This is great!

Mind if we rename some things?


- `noInput` -> `run`
- `simple` -> `fromJson`
- `simpleFrom` -> `fromText`
- `run` -> `withFeedback`

Reply:
Go for it, the PR should allow pushes from you